### ISSUE TITLE
Required fields are force-fed stars?

### ISSUE THREAD

**Issue by nternetinspired** [joomla/joomla-cms#4139]
# Why stars?
Showing a star next to a required field is s stylistic treatment (possibly also meaningless in many cultures) with *zero functional benefit*. A browser will consider it as meaningful as a :poop: symbol, it has no place in markup.

Nonetheless, if a form input is defined as required (I'm looking at com_users currently) in the forms/*.xml it is output with a meaningless * appended to the `label`. 

    <label id="password-lbl" for="password" class=" required">Password<span class="star">&nbsp;*</span></label>

## Why not stars?
This is enforcing a style trump over the wishes of template authors, site owners and end users and it is nuts. It can't be overridden, AFAIK. I would have to hack core to remove this star symbol.

What if I don't want a star on every required form field? What if I want to use another icon? What if I want a clown-face or an :8ball: ??

## Badges? We don't need no steenkin' badges
A wholly stylistic treatment like this should only *ever* be applied in the presentation layer, i.e. the template. The current behaviour isn't correct in any sense, semantically or stylistically. If I knew how to correct this I would submit a PR in an instant, unfortunately I don't. I'm restricted, as a front-ender, to;

* Hiding .star in css, an ugly hack.
* Hacking core to remove the required attribute, an even uglier hack.
* Creating rage-fuelled Github issues.

I can't think of any justifiable reason for this behaviour but if anyone can I'd love to hear it. I promise to remain calm!

In summary: 

`<label class="required">` = good.
`<span class="star">&nbsp;*</span>` = bad.

**Comment by brianteeman** [comment]
What is the status of this? Is it to be closed for the B/C issues or is someone going to create the PR with the deprecated notice as @nicksavov requested or something else?<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4139">issues.joomla.org/joomla-cms/4139</a>.</sub>

**Comment by nternetinspired** [comment]
So, to re-iterate for clarity, are we are all agreed that:

* All instances of `<span class="star">&nbsp;*</span>` in rendered markup should be marked as deprecated for 4.0.
* Default templates for 4.0 will use pure CSS implementations to insert * as CSS content?

?

**Comment by brianteeman** [comment]
1. agreed from me
2. agreed from me but I guess as no plans have been done for 4.x yet that
can only be an aim and not a firm guarantee

On 5 January 2015 at 09:42, Seth Warburton <notifications@github.com> wrote:

> So, to re-iterate for clarity, are we are all agreed that:
>
>    - All instances of <span class="star">&nbsp;*</span> in rendered
>    markup should be marked as deprecated for 4.0.
>    - Default templates for 4.0 will use pure CSS implementations to
>    insert * as CSS content?
>
> ?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/joomla/joomla-cms/issues/4139#issuecomment-68686005>.
>



-- 
Brian Teeman
Co-founder Joomla! and OpenSourceMatters Inc.
http://brian.teeman.net/

**Comment by infograf768** [comment]
Note: We also have `<span class="star">&#160;*</span>`

**Comment by brianteeman** [comment]
@nternetinspired are you going to submit a PR that marks this as deprecated or shall I just close this<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/4139">issues.joomla.org/joomla-cms/4139</a>.</sub>

**Comment by zero-24** [comment]
@brianteeman can you add the `Reevaluate for v4.0` label so this will be on the list for 4.0?

**Comment by nternetinspired** [comment]
@brianteeman @nicksavov It would be imprudent to mark these instances as deprecated when there seems to be no clear direction on the template issue, wouldn't it?

If we are to stay with :star:'s in the near future then any markup changes really need to occur hand-in-hand with a template-based alternative.

**Comment by bluezeyes** [comment]
From my point of view as end user and beginner, I would prefer to remove that span * asap as from one perspective in this discussion that point was not mentioned is, that all users which uses screenreaders have with the current solution a problem that they only "hear" The field title and the content "star" which has nothing todo with the required aspect of that field.

Instead, using the required atribute, is help those programs to give more information about what the user have to fill out, give them a chance to optimize the Layout on client site, etc Those attributes like input type="[number|date|etc]" in HTML5 , more gives the Browser and mobile clients a chance to change to right input layout, check the validity and range.. To increase the accessibility for disabled visitors take more advance than just leaving that span * for BC... 

Just my 2c

**Comment by nternetinspired** [comment]
@bluezeyes Thank you! This is probably the best argument presented in this thread, * is not an accessible solution to the problem of informing a user about required fields.

**Comment by zero-24** [comment]
Closing as this issue have the Re-evaluate for v4.0 label :smile:
